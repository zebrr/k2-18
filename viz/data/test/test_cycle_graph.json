{
  "nodes": [
    {
      "id": "chunk_a",
      "type": "Chunk",
      "text": "Node A: First node in the cycle",
      "node_offset": 0,
      "definition": "First node in cyclic test graph",
      "difficulty": 2
    },
    {
      "id": "chunk_b",
      "type": "Chunk",
      "text": "Node B: Second node in the cycle",
      "node_offset": 100,
      "definition": "Second node in cyclic test graph",
      "difficulty": 2
    },
    {
      "id": "chunk_c",
      "type": "Chunk",
      "text": "Node C: Third node in the cycle",
      "node_offset": 200,
      "definition": "Third node in cyclic test graph",
      "difficulty": 2
    },
    {
      "id": "chunk_d",
      "type": "Chunk",
      "text": "Node D: Fourth node in the cycle",
      "node_offset": 300,
      "definition": "Fourth node in cyclic test graph",
      "difficulty": 2
    }
  ],
  "edges": [
    {
      "source": "chunk_a",
      "target": "chunk_b",
      "type": "PREREQUISITE",
      "weight": 1.0,
      "conditions": "A connects to B in the cycle"
    },
    {
      "source": "chunk_b",
      "target": "chunk_c",
      "type": "PREREQUISITE",
      "weight": 1.0,
      "conditions": "B connects to C in the cycle"
    },
    {
      "source": "chunk_c",
      "target": "chunk_d",
      "type": "PREREQUISITE",
      "weight": 1.0,
      "conditions": "C connects to D in the cycle"
    },
    {
      "source": "chunk_d",
      "target": "chunk_a",
      "type": "PREREQUISITE",
      "weight": 1.0,
      "conditions": "D connects back to A, completing the cycle"
    }
  ]
}
