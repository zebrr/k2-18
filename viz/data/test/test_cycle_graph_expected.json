{
  "nodes": [
    {
      "id": "chunk_a",
      "type": "Chunk",
      "text": "Node A: First node in the cycle",
      "node_offset": 0,
      "definition": "First node in cyclic test graph",
      "difficulty": 2,
      "degree_in": 1,
      "degree_out": 1,
      "degree_centrality": 0.6667,
      "pagerank": 0.25,
      "betweenness_centrality": 0.5,
      "out-closeness": 0.5,
      "component_id": 0,
      "prerequisite_depth": 0,
      "learning_effort": 8,
      "educational_importance": 0.250000
    },
    {
      "id": "chunk_b",
      "type": "Chunk",
      "text": "Node B: Second node in the cycle",
      "node_offset": 100,
      "definition": "Second node in cyclic test graph",
      "difficulty": 2,
      "degree_in": 1,
      "degree_out": 1,
      "degree_centrality": 0.6667,
      "pagerank": 0.25,
      "betweenness_centrality": 0.5,
      "out-closeness": 0.5,
      "component_id": 0,
      "prerequisite_depth": 0,
      "learning_effort": 8,
      "educational_importance": 0.250000
    },
    {
      "id": "chunk_c",
      "type": "Chunk",
      "text": "Node C: Third node in the cycle",
      "node_offset": 200,
      "definition": "Third node in cyclic test graph",
      "difficulty": 2,
      "degree_in": 1,
      "degree_out": 1,
      "degree_centrality": 0.6667,
      "pagerank": 0.25,
      "betweenness_centrality": 0.5,
      "out-closeness": 0.5,
      "component_id": 0,
      "prerequisite_depth": 0,
      "learning_effort": 8,
      "educational_importance": 0.250000
    },
    {
      "id": "chunk_d",
      "type": "Chunk",
      "text": "Node D: Fourth node in the cycle",
      "node_offset": 300,
      "definition": "Fourth node in cyclic test graph",
      "difficulty": 2,
      "degree_in": 1,
      "degree_out": 1,
      "degree_centrality": 0.6667,
      "pagerank": 0.25,
      "betweenness_centrality": 0.5,
      "out-closeness": 0.5,
      "component_id": 0,
      "prerequisite_depth": 0,
      "learning_effort": 8,
      "educational_importance": 0.250000
    }
  ],
  "edges": [
    {
      "source": "chunk_a",
      "target": "chunk_b",
      "type": "PREREQUISITE",
      "weight": 1.0,
      "inverse_weight": 1.0,
      "conditions": "A connects to B in the cycle"
    },
    {
      "source": "chunk_b",
      "target": "chunk_c",
      "type": "PREREQUISITE",
      "weight": 1.0,
      "inverse_weight": 1.0,
      "conditions": "B connects to C in the cycle"
    },
    {
      "source": "chunk_c",
      "target": "chunk_d",
      "type": "PREREQUISITE",
      "weight": 1.0,
      "inverse_weight": 1.0,
      "conditions": "C connects to D in the cycle"
    },
    {
      "source": "chunk_d",
      "target": "chunk_a",
      "type": "PREREQUISITE",
      "weight": 1.0,
      "inverse_weight": 1.0,
      "conditions": "D connects back to A, completing the cycle"
    }
  ]
}
